sympy/matrices/matrices.py

```diff
@@ -1495,6 +1496,189 @@ def is_diagonalizable(self, reals_only=False, **kwargs):
                 ret = False
         return ret
 
+    def _eval_is_positive_definite(self, method="eigen"):
+        """Algorithm dump for computing positive-definiteness of a
+        matrix.
+
+        Parameters
+        ==========
+
+        method : str, optional
+            Specifies the method for computing positive-definiteness of
+            a matrix.
+
+            If ``'eigen'``, it computes the full eigenvalues and decides
+            if the matrix is positive-definite.
+
+            If ``'CH'``, it attempts computing the Cholesky
+            decomposition to detect the definitiveness.
+
+            If ``'LDL'``, it attempts computing the LDL
+            decomposition to detect the definitiveness.
+        """
+        if self.is_hermitian:
+            if method == 'eigen':
+                eigen = self.eigenvals()
+                args = [x.is_positive for x in eigen.keys()]
+                return fuzzy_and(args)
+
+            elif method == 'CH':
+                try:
+                    self.cholesky(hermitian=True)
+                except ValueError as e:
+                    if str(e) == "Matrix must be positive-definite":
+                        return False
+                    raise ValueError()
```

Comment:
Testing for the string in the exception message seems flaky to me. Maybe those methods should return a subclass of ValueError like NotPositiveDefiniteError that could be caught here.

Also it would be better to use just `raise` rather than `raise ValueError()` to preserve the original exception message and traceback.

(same comments for below)